### PR TITLE
Add FXIOS-3981 [v102] - Add default cases to ViewModel: Wallpapers, Home page settings, Sync

### PR DESCRIPTION
**Pull request by yoanarios**
Includes FXIOS-3981 - Creation of reusable component for Onboarding
 
 - Add PageViewController to swipe between onboarding screens
 - Add vertical scroll view to  support dynamic  font
 - Add functionality to set wallpapers and start sing-in for Sync Account
 - Handle default onboarding flow
 - Unit test

### CODE REVIEW

File: Client/Frontend/Intro/IntroViewController.swift [hunk 1]
```diff
@@ -46,68 +73,147 @@ class IntroViewController: UIViewController, OnViewDismissable {
...
+        moveToNextPage(cardType: cardType)
+    }
+
+    func primaryAction(_ cardType: IntroViewModel.OnboardingCards) {
+        switch cardType {
+        case .welcome, .wallpapers:
+            moveToNextPage(cardType: cardType)
+        case .signSync:
+            presentSignToSync()
+
```
Comment (nbhasin2) [inline Client/Frontend/Intro/IntroViewController.swift:190]: nit: extra line

```suggestion
```

File: Client/Frontend/Intro/IntroViewController.swift [hunk 2]
```diff
@@ -46,68 +73,147 @@ class IntroViewController: UIViewController, OnViewDismissable {
...
+        var cardViewController: OnboardingCardViewController
+        for (index, cardType) in viewModel.enabledCards.enumerated() {
+            if cardType == .wallpapers {
+                cardViewController = WallpaperCardViewController(viewModel: viewModel.getCardViewModel(index: index),
+                                                                      delegate: self)
+            } else {
+                cardViewController = OnboardingCardViewController(viewModel: viewModel.getCardViewModel(index: index),
+                                                                      delegate: self)
+            }
+            onboardingCards.append(cardViewController)
```
Comment (nbhasin2) [inline Client/Frontend/Intro/IntroViewController.swift:87]: Indentation seems off on XCode and github 

[Before]

image

This is the best I feel we can fix it [after] 

image

File: Client/Frontend/Intro/IntroViewController.swift [hunk 3]
```diff
@@ -8,25 +8,49 @@ import Shared
...
-        return welcomeCardView
-    }()
-    private lazy var syncCard: IntroScreenSyncView = {
-        let syncCardView = IntroScreenSyncView()
-        syncCardView.translatesAutoresizingMaskIntoConstraints = false
-        syncCardView.clipsToBounds = true
-        return syncCardView
+    var viewModel: IntroViewModel
+    let profile: Profile
+    var onboardingCards = [OnboardingCardViewController]()
```
Comment (nbhasin2) [inline Client/Frontend/Intro/IntroViewController.swift:13]: nit: 
```suggestion
    private var viewModel: IntroViewModel
    private let profile: Profile
    private var onboardingCards = [OnboardingCardViewController]()
```

File: Client/Frontend/Intro/OnboardingCardViewController.swift [hunk 4]
```diff
@@ -0,0 +1,229 @@
...
+
+            buttonStackView.leadingAnchor.constraint(equalTo: containerView.leadingAnchor, constant: UX.stackViewPadding),
+            buttonStackView.bottomAnchor.constraint(equalTo: containerView.bottomAnchor, constant: -20),
+            buttonStackView.trailingAnchor.constraint(equalTo: containerView.trailingAnchor, constant: -UX.stackViewPadding),
+
+            primaryButton.heightAnchor.constraint(greaterThanOrEqualToConstant: UX.buttonHeight),
+            secondaryButton.heightAnchor.constraint(greaterThanOrEqualToConstant: UX.buttonHeight)
+        ])
+
+//        containerView.setContentHuggingPriority(.defaultHigh, for: .vertical)
```
Comment (nbhasin2) [inline Client/Frontend/Intro/OnboardingCardViewController.swift:197]: nit: unused code / cleanup 🤔 
```suggestion
```

**Review by nbhasin2 — APPROVED** [review]

**Comment by nbhasin2** [comment]
Overall looks good 🚀

**Comment by yoanarios** [comment]
Failing test are the ones related to this issue #10902

**Pull request merged**